siliconcompiler/schema/editableschema.py

```
@@ -146,3 +152,26 @@ def search(self, *keypath: str) -> Union[BaseSchema, Parameter]:
+        new_schema = self.__schema.copy()
+        if new_schema._parent() is not new_schema:
+            new_schema._BaseSchema__parent = None
+        return new_schema
+
+    def rename(self, name: str):
+        '''
+        Renames a named schema
+        '''
+
+        if not isinstance(self.__schema, NamedSchema):
+            raise TypeError("schema must be a named schema")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)